Automattic/pocket-casts-android
# Adds a cancel confirmation view

## Description
Adds a cancel confirmation view. 

iOS PR: https://github.com/Automattic/pocket-casts-ios/pull/591

## Testing Instructions

1. Apply patch to test free trial subscription (p1668803843029399-slack-C028JAG44VD)
2. Launch the app
3. Sign into an account without plus and subscribe to plus from the app
4. Tap Profile tab
5. Tap your account
6. Tap Cancel subscription
7. ✅ Verify you see the following in console: `🔵 Tracked: account_details_cancel_tapped`
8. ✅ Verify you see the following in console: `🔵 Tracked: cancel_confirmation_view_shown`
9. ✅ Verify the view looks good
10. Swipe down to dismiss
11. ✅ Verify you see the following in console: `🔵 Tracked: cancel_confirmation_view_dismissed`
12. Reopen the view
13. Tap the stay button
14. ✅ Verify you see the following in console: `🔵 Tracked: cancel_confirmation_stay_button_tapped`
15. Reopen the view, and tap cancel 
16. ✅ Verify you see the following in console: `🔵 Tracked: cancel_confirmation_cancel_button_tapped`

## Screenshots or Screencast 
<img width=320 src="https://user-images.githubusercontent.com/1405144/207285765-c4602f23-9a02-4be5-afbc-9ec99cb27a1c.png"/>


## Checklist
- If this is a user-facing change, I have added an entry in CHANGELOG.md
- I have considered whether it makes sense to add tests for my changes
 
#### I have tested any UI changes...
